packages/frontend/src/components/Swap.tsx

```diff
@@ -78,9 +80,17 @@ export default function Trade({ handleOpenModal }: Props) {
   async function swapTokens({ token, account, value }: any) {
     if (value && token && parseFloat(tokenBalance) >= value) {
       try {
-        await swap(token!, account!, value!.toString());
+        setTxLoading(true);
+        await swap(token!, account!, value!.toString()).then((data) => {
```

Comment:
change to
const data = await swap(token!, account!, value!.toString());
no need to use .then() here